src/pages/Settings.vue

```diff
@@ -121,6 +147,28 @@ export default {
                 },
             };
         },
+
+        breadcrumbs: ({ $route, subMenus }) => {
```

Comment:
This function seems weird and I don't fully understand what it does.
Do you think extracting some parts to another computed property would be benefitial for readablity?

Reply:
This computed produce an array of routes, in the right order to generate a breadcrumb related to a setting. This is something new, needed for the users management. You can see it in action is this screenshot.
![image](https://github.com/M1CK431/uptime-kuma/assets/31622263/3a882077-8d09-4823-94ef-c4e89f5bf654)

I agree that code is a bit complicated... but it does something complicated! Do you want some explanation about how it works and why it is needed?

>Do you think extracting some parts to another computed property would be benefitial for readablity?

Not sure, but I could add as many comments as needed to make things clearer? 🤷🏼‍♂️ 
I could also extract some part in dedicated helper function?

Reply:
I would argue that splitting this into `uniqueSettingRoutes` and `breadcrumbs` would be more readable.
Also that map-reduce is quite complicated (the title especially).

Inlining `uniqBy` could also reduce the complexity of the function:
- first map to list of strings (`path`s),
- filter the results
- return `[...new Set(filteredPaths)]`

Reply:
There is performance consideration to take into account here.

> I would argue that splitting this into uniqueSettingRoutes and breadcrumbs would be more readable.

Be careful: in VueJS, each computed will produce a "watcher" for all computed getter function dependencies, so the more computed you add, the more watcher VueJS will create for you. [Too many watchers are bad from performances point of view](https://vuejs.org/guide/essentials/watchers.html#deep-watchers). At the end, spliting it in multiple computed might have a (very small) performance impact.

> Also that map-reduce is quite complicated (the title especially).

Please excuse me if I'm wrong, but I have the feeling that you are not very familiar with [JS destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment)? I'm just extracting the `title` from the returned objet, nothing really complicated I guess? 🤔😮 

> irst map to list of strings (paths),
> filter the results
> return [...new Set(filteredPaths)]

Sorry, I can't apply that suggestion because it's not what happens in the current implementation.
I will extract some part in helpers functions and add more comments, then I will request your review to be sure that the new code is easier to understand/maintain 😘

Reply:
@CommanderStorm could you please take a look to that commit: https://github.com/M1CK431/uptime-kuma/pull/1/commits/50fd38b820a6d8e529c10747e6c5e8b679511a2e

I add A LOT of comments with examples and I extract the `reduce` part in a documented helper function.
Is this clear enough now?

Reply:
Resolved